File: app/src/main/java/com/chirag/nasaphotooftheday/misc/AppAnimationUtils.java
package com.chirag.nasaphotooftheday.misc;

import android.view.View;
import android.view.animation.AccelerateInterpolator;
import android.view.animation.AlphaAnimation;
import android.view.animation.Animation;

/**
 * Created by Chirag Desai
 */
public class AppAnimationUtils {

    public static void fadeInAnimation(final View view, long animationDuration, int startOffset) {
        Animation fadeIn = new AlphaAnimation(0, 1);
        fadeIn.setInterpolator(new AccelerateInterpolator());
        fadeIn.setDuration(animationDuration);
        fadeIn.setStartOffset(startOffset);
        fadeIn.setAnimationListener(new Animation.AnimationListener() {
            @Override
            public void onAnimationStart(Animation animation) {
            }

            @Override
            public void onAnimationEnd(Animation animation) {
                view.setVisibility(View.VISIBLE);
            }

            @Override
            public void onAnimationRepeat(Animation animation) {
            }
        });

        view.startAnimation(fadeIn);
    }
}

File: app/src/main/java/com/chirag/nasaphotooftheday/misc/AppAnimationUtils.kt
package com.chirag.nasaphotooftheday.misc

import android.view.View
import android.view.animation.AccelerateInterpolator
import android.view.animation.AlphaAnimation
import android.view.animation.Animation

/**
 * Created by Chirag Desai
 */
object AppAnimationUtils {
    fun fadeInAnimation(
        view: View,
        animationDuration: Long,
        startOffset: Int
    ) {
        val fadeIn: Animation = AlphaAnimation(0f, 1f)
        fadeIn.interpolator = AccelerateInterpolator()
        fadeIn.duration = animationDuration
        fadeIn.startOffset = startOffset.toLong()
        fadeIn.setAnimationListener(object : Animation.AnimationListener {
            override fun onAnimationStart(animation: Animation) {}
            override fun onAnimationEnd(animation: Animation) {
                view.visibility = View.VISIBLE
            }

            override fun onAnimationRepeat(animation: Animation) {}
        })
        view.startAnimation(fadeIn)
    }
}